thorntail-runner/src/main/java/org/wildfly/swarm/runner/WarBuilder.java

```diff
@@ -54,9 +55,40 @@ private WarBuilder(ZipOutputStream output, List<String> classesDirs, List<File>
     }
 
     private void build() {
-        classesDirs.forEach(this::addClassesToWar);
+        List<String> warDirs = classesDirs.stream()
+                .filter(this::isWar)
+                .collect(Collectors.toList());
+
+        classesDirs.stream()
+                .filter(d -> !isWar(d))
+                .map(this::buildJar)
+                .forEach(jars::add);
+
+        warDirs.forEach(this::addClassesToWar);
         addWebAppResourcesToWar();
-        jars.stream().forEach(this::addJarToWar);
+
+        jars.forEach(this::addJarToWar);
+    }
+
+    private File buildJar(String classesDir) {
+        try {
+            File jar = File.createTempFile("thorntail-user-jar", ".jar");
+
+            try (FileOutputStream fos = new FileOutputStream(jar);
+                 ZipOutputStream out = new ZipOutputStream(fos)) {
+                addClassesToXar(out, "", new File(classesDir));
+            }
+            return jar;
+        } catch (Exception any) {
+            throw new RuntimeException("Unable to build jar from directory: " + classesDir);
+        }
+    }
+
+    // the assumption is that the Runner is invoked in the WAR module's directory
```

Comment:
I wonder if that assumption makes sense. WAR could possibly be detected by presence of `src/main/webapp` or some other means. And we should perhaps detect if the found more than 1 WAR and bail out in that case. WDYT?

Reply:
To have all the dependencies visible for `Runner`, one should run the project from the war module. That was my reasoning.

Reply:
Ah OK. Is it possible to detect if the `Runner` was executed from another module and offer an error message? Just thinking out loud.

Reply:
I think it would be best to figure it out from Maven or Gradle files. Otherwise I don't see an easy way. 
Otherwise we may check if webapp dir is there but I think it's perfectly okay not to have it.

I added another bullet to https://issues.jboss.org/browse/THORN-2244 regarding figuring out what is a `WAR` and what's not.

Reply:
Cool, thanks.